test/e2e/framework/pod/wait.go

```diff
@@ -856,3 +856,14 @@ func WaitForContainerTerminated(ctx context.Context, c clientset.Interface, name
 		return false, nil
 	})
 }
+
+func GetPodNodeName(ctx context.Context, c clientset.Interface, ns, name string) (string, error) {
+	pod, err := c.CoreV1().Pods(ns).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	if pod.Spec.NodeName == "" {
+		return "", fmt.Errorf("pod %s/%s has no node name", ns, name)
+	}
+	return pod.Spec.NodeName, nil
+}
```

Comment:
remove this function. Not needed IMO.

Reply:
The reason is mentioned in the previous comment, I think we should keep the current method since I haven't found any alternative method.

Reply:
I do not understand, why this function will be needed. If a function is returning a pod without node name being set after creation, then that function should be fixed?

Reply:
The specific function was cities in [code](https://github.com/kubernetes/kubernetes/blob/5a99930ab57b2c7837960b21d2c92e958db8c377/test/e2e/storage/csimock/base.go#L735) and it's context. IMHO, Modifying this method might have implications on other e2e tests, suggesting that it could be wiser to implement a separate method. Considering pod creation and scheduling are inherently asynchronous, it's not ideal to wait during the creation process.

Reply:
Please don't add this to `e2e/framework/pod`, in particular without any documentation. It's not clear why this needs to be here.

If you think that your test benefits from having such a helper function, then add it in `csi_node_volume_health.go`. But to be honest, it might be easier to read the test when not using such a helper.

Reply:
@pohly  comment fixed, ptal